x-pack/plugins/security_solution/public/overview/containers/overview_cti_links/use_ti_data_sources.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Observable } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { useEffect, useState } from 'react';
+import { useObservable, withOptionalSignal } from '@kbn/securitysolution-hook-utils';
+import { useKibana } from '../../../common/lib/kibana';
+import {
+  DataPublicPluginStart,
+  isCompleteResponse,
+  isErrorResponse,
+} from '../../../../../../../src/plugins/data/public';
+import {
+  Bucket,
+  CtiQueries,
+  CtiDataSourceStrategyResponse,
+  CtiDataSourceRequestOptions,
+} from '../../../../common';
+import { DEFAULT_THREAT_INDEX_KEY } from '../../../../common/constants';
+import { GlobalTimeArgs } from '../../../common/containers/use_global_time';
+import { OTHER_DATA_SOURCE_TITLE } from '../../components/overview_cti_links/translations';
+import { OTHER_TI_DATASET_KEY } from '../../../../common/cti/constants';
+
+type GetThreatIntelSourcProps = CtiDataSourceRequestOptions & {
+  data: DataPublicPluginStart;
+  signal: AbortSignal;
+};
+export const ID = 'ctiEventCountQuery';
+
+export const getTiDataSources = ({
+  data,
+  defaultIndex,
+  timerange,
+  signal,
+}: GetThreatIntelSourcProps): Observable<CtiDataSourceStrategyResponse> =>
+  data.search.search<CtiDataSourceRequestOptions, CtiDataSourceStrategyResponse>(
+    {
+      defaultIndex,
+      factoryQueryType: CtiQueries.dataSource,
+      timerange,
+    },
+    {
+      strategy: 'securitySolutionSearchStrategy',
+      abortSignal: signal,
+    }
+  );
+
+export const getTiDataSourcesComplete = (
+  props: GetThreatIntelSourcProps
+): Observable<CtiDataSourceStrategyResponse> => {
+  return getTiDataSources(props).pipe(
+    filter((response) => {
+      return isErrorResponse(response) || isCompleteResponse(response);
+    })
+  );
+};
+
+const getTiDataSourcesWithOptionalSignal = withOptionalSignal(getTiDataSourcesComplete);
+
+export const useTiDataSourcesComplete = () => useObservable(getTiDataSourcesWithOptionalSignal);
+
+export interface TiDataSources {
+  dataset: string;
+  name: string;
+  count: number;
+  dashboardId?: string;
+}
+interface TiDataSourcesProps extends Partial<GlobalTimeArgs> {
+  allTiDataSources?: TiDataSources[];
```

Comment:
I believe this prop can be made non-optional, no?
```suggestion
  allTiDataSources: TiDataSources[];
```

Reply:
Not really, we have a case when don't pass it, because we don't have it yet
`x-pack/plugins/security_solution/public/overview/containers/overview_cti_links/use_all_ti_data_sources.ts`